### PR TITLE
API Updates

### PR DESCRIPTION
**Pull request by richardm-stripe**
Codegen for openapi 3c477c0.
r? @richardm-stripe
cc @stripe/api-libraries

## Changelog
* Add support for `konbini_payments` on `AccountUpdateParams.capabilities`, `AccountCreateParams.capabilities`, and `Account.capabilities`
* Add support for `konbini` on `Charge.payment_method_details`, `CheckoutSessionCreateParams.payment_method_options`, `Checkout.Session.payment_method_options`, `InvoiceCreateParams.payment_settings.payment_method_options`, `InvoiceUpdateParams.payment_settings.payment_method_options`, `Invoice.payment_settings.payment_method_options`, `PaymentIntentCreateParams.payment_method_data`, `PaymentIntentCreateParams.payment_method_options`, `PaymentIntentUpdateParams.payment_method_data`, `PaymentIntentUpdateParams.payment_method_options`, `PaymentIntentConfirmParams.payment_method_data`, `PaymentIntentConfirmParams.payment_method_options`, `PaymentIntent.payment_method_options`, `PaymentMethodCreateParams`, `PaymentMethod`, `SubscriptionCreateParams.payment_settings.payment_method_options`, `SubscriptionUpdateParams.payment_settings.payment_method_options`, and `Subscription.payment_settings.payment_method_options`
* Add support for new value `konbini` on enums `CheckoutSessionCreateParams.payment_method_types[]` and `PaymentMethodCreateParams.type`
* Add support for new value `konbini` on enums `CustomerListPaymentMethodsParams.type` and `PaymentMethodListParams.type`
* Add support for new value `konbini` on enums `InvoiceCreateParams.payment_settings.payment_method_types[]`, `InvoiceUpdateParams.payment_settings.payment_method_types[]`, `SubscriptionCreateParams.payment_settings.payment_method_types[]`, and `SubscriptionUpdateParams.payment_settings.payment_method_types[]`
* Add support for new value `konbini` on enums `PaymentIntentCreateParams.payment_method_data.type`, `PaymentIntentUpdateParams.payment_method_data.type`, and `PaymentIntentConfirmParams.payment_method_data.type`
* Add support for `konbini_display_details` on `PaymentIntent.next_action`

### CODE REVIEW

File: src/main/java/com/stripe/model/PaymentIntent.java [hunk 1]
```diff
@@ -1308,6 +1311,59 @@ public void setPackage(String package_) {
...
+    String hostedVoucherUrl;
+
+    @SerializedName("stores")
+    NextActionKonbiniDisplayDetailsStores stores;
+  }
+
+  @Getter
+  @Setter
+  @EqualsAndHashCode(callSuper = false)
+  public static class NextActionKonbiniDisplayDetailsStores extends StripeObject {
```
Comment (richardm-stripe) [inline src/main/java/com/stripe/model/PaymentIntent.java:1336]: This is almost certainly not what we want. Should be just `public static class Stores` and be an inner class of `NextActionKonbiniDisplayDetails`

File: src/main/java/com/stripe/model/PaymentIntent.java [hunk 2]
```diff
@@ -1308,6 +1311,59 @@ public void setPackage(String package_) {
...
+
+    /** Seicomart instruction details. */
+    @SerializedName("seicomart")
+    NextActionKonbiniDisplayDetailsStoresStore seicomart;
+  }
+
+  @Getter
+  @Setter
+  @EqualsAndHashCode(callSuper = false)
+  public static class NextActionKonbiniDisplayDetailsStoresStore extends StripeObject {
```
Comment (richardm-stripe) [inline src/main/java/com/stripe/model/PaymentIntent.java:1357]: This should also be an inner class.

To be really consistent, each of these would be separate classes, too...

File: src/main/java/com/stripe/model/Subscription.java [hunk 3]
```diff
@@ -824,6 +824,13 @@ public Discount deleteDiscount(Map<String, Object> params, RequestOptions option
     @SerializedName("card")
     Card card;
 
+    /**
+     * This sub-hash contains details about the Konbini payment method options to pass to invoices
+     * created by the subscription.
+     */
+    @SerializedName("konbini")
+    Invoice.PaymentMethodOptions.Konbini konbini;
```
Comment (richardm-stripe) [inline src/main/java/com/stripe/model/Subscription.java:832]: We're going to have to unshare this.

File: src/main/java/com/stripe/model/SetupIntentTypeSpecificPaymentMethodOptionsClient.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+package com.stripe.model;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@EqualsAndHashCode(callSuper = false)
+public class SetupIntentTypeSpecificPaymentMethodOptionsClient extends StripeObject {}
```
Comment (richardm-stripe) [inline src/main/java/com/stripe/model/SetupIntentTypeSpecificPaymentMethodOptionsClient.java:11]: Oof this should not be generated.

**Review by remi-stripe — APPROVED** [review]
LGTM

**Pull request merged**